hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamerUtils.java

```diff
@@ -159,7 +156,16 @@ public static Option<JavaRDD<HoodieRecord>> createHoodieRecords(HoodieStreamer.C
    * @return the representation of error record (empty {@link HoodieRecord} and the error record
    * String) for writing to error table.
    */
-  private static Either<HoodieRecord, String> generateErrorRecord(GenericRecord genRec) {
+  private static Either<HoodieRecord, String> generateErrorRecordOrThrowException(GenericRecord genRec, Exception e, boolean shouldErrorTable) {
+    if (!shouldErrorTable) {
+      if (e instanceof HoodieKeyException) {
+        throw (HoodieKeyException) e;
+      } else if (e instanceof HoodieKeyGeneratorException) {
+        throw (HoodieKeyGeneratorException) e;
```

Comment:
Nit: could the `if` and `else if` be merged like before?  Also, there is no need to do type cast here.

Reply:
I thought that too. But that was the only way I could get it to compile.

Reply:
Got it